csrc/pf_core.c

```diff
@@ -223,6 +223,10 @@ PForthDictionary pfCreateDictionary( cell_t HeaderSize, cell_t CodeSize )
 
     dic->dic_Flags |= PF_DICF_ALLOCATED_SEGMENTS;
```

Comment:
We are setting the flag to say we have allocated the segments. But then we return before allocating them.